### PR TITLE
Bugfix for donation usdValue set

### PR DESCRIPTION
**Pull request by arbreton**
Research on Donation Hooks usdValue setter

### CODE REVIEW

**Comment by arbreton** [comment]
@aminlatifi eslint should be fixed now

File: src/services/donations/donations.hooks.js [hunk 1]
```diff
@@ -117,15 +117,21 @@ const setUSDValue = async (context, donation) => {
     const { rate } = await getHourlyUSDCryptoConversion(context.app, createdAt, token.symbol);
 
     if (rate) {
+      let usVar = 0;
+      if (token.symbol === 'PAN') {
+        usVar = rate;
```
[the remark this answers is not in the document]
Comment (arbreton) [reply src/services/donations/donations.hooks.js:122]: What was the question? I got confused a little, 
There seems to be a different result from coingecko and cryptocompare, so basically coingecko returns a rate and cryptocompare needs a conversion of some sort 
Oh i think i understand what you are saying!
Yeah basically coingecko returns the rate flat thats why when it's pan the rate is used for usdVar

File: src/services/conversionRates/getConversionRatesService.js [hunk 2]
```diff
@@ -143,6 +143,64 @@ const _getRatesCryptocompare = async (timestamp, ratesToGet, symbol, stableCoins
...
+      bestPrice = price;
+    } else {
+      bestPrice = 1;
+    }
+    rate = bestPrice;
+  } else {
+    rate = 1;
+  }
+
+  return rate.toFixed(2);
```
[the remark this answers is not in the document]
Comment (arbreton) [reply src/services/conversionRates/getConversionRatesService.js:201]: Should we try 4-5 maybe? i think that could be better?

File: src/services/donations/donations.hooks.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (arbreton) [reply src/services/donations/donations.hooks.js]: Should be fixed now, it should be related to amount

File: src/services/conversionRates/getConversionRatesService.js [hunk 3]
```diff
@@ -143,6 +143,64 @@ const _getRatesCryptocompare = async (timestamp, ratesToGet, symbol, stableCoins
...
+      bestPrice = price;
+    } else {
+      bestPrice = 1;
+    }
+    rate = bestPrice;
+  } else {
+    rate = 1;
+  }
+
+  return rate.toFixed(5);
```
Comment (aminlatifi) [inline src/services/conversionRates/getConversionRatesService.js:201]: I meant no `toFixed` here.

File: src/services/donations/donations.hooks.js [hunk 4]
```diff
@@ -117,15 +117,16 @@ const setUSDValue = async (context, donation) => {
     const { rate } = await getHourlyUSDCryptoConversion(context.app, createdAt, token.symbol);
 
     if (rate) {
+      const usVar = Number(
+        new BigNumber(amount)
+          .div(10 ** 18)
+          .times(rate)
+          .toFixed(5),
```
Comment (aminlatifi) [inline src/services/donations/donations.hooks.js:124]: Revert it to same `.toFixed(2)`, I didn't say to change it.

**Review by aminlatifi — CHANGES_REQUESTED** [review]
@arbreton Regardless of mentioned problems, I think it should be tested once with real data. As talked on last meeting port it to `scripts/utils/DonationUsdValueUtility.js` and test it with our real Pan donation values

File: src/services/conversionRates/getConversionRatesService.js (hunk 3 above)
Comment (arbreton) [reply src/services/conversionRates/getConversionRatesService.js:201]: I will remove toFixed

File: src/services/donations/donations.hooks.js (hunk 4 above)
Comment (arbreton) [reply src/services/donations/donations.hooks.js:124]: Will revert to` .toFixed(2)`

**Comment by aminlatifi** [comment]
It seems OK

**Pull request merged**